de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/console/ServerConsole.java

```diff
@@ -0,0 +1,68 @@
+package de.fu_berlin.inf.dpp.server.console;
+
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * Console implementation for the SarosServer.
+ *
+ * Dynamically handles user commands and executes
+ * registered {@link ConsoleCommand}s.
+ */
+public class ServerConsole implements Runnable {
+    private Scanner input;
+    private PrintStream output;
+    private List<ConsoleCommand> commands;
+
+    public ServerConsole(InputStream in, PrintStream out) {
+        input = new Scanner(in);
+        output = out;
+        commands = new ArrayList<>();
+    }
+
+    public void registerCommand(ConsoleCommand command) {
+        commands.add(command);
+    }
+
+    @Override
+    public void run() {
+        output
+            .println("# Welcome to Saros Server (type 'help' for available commands)");
+        while (true) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Reduce the total number of break and continue statements in this loop to use at most one. ([squid:S135](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS135))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=143&issues=AWZTTLKuBfefbO9HkYcf&open=AWZTTLKuBfefbO9HkYcf)

[ISSUE_KEY]: AWZTTLKuBfefbO9HkYcf